lib/elixir/lib/string.ex

```diff
@@ -709,6 +709,34 @@ defmodule String do
     {left, right}
   end
 
+  @doc ~S"""
+  Counts the number of occurrences of a pattern in a string.
+
+  ## Examples
+
+      iex> String.count("hello world", "o")
+      2
+
+      iex> String.count("hello world", "l")
+      3
+
+      iex> String.count("hello world", "x")
+      0
+
+      iex> String.count("hello world", ~r/o/)
+      2
+
+  """
+  @spec count(t, pattern | Regex.t()) :: non_neg_integer
+  @doc since: "1.19.0"
+  def count(string, pattern) when is_struct(pattern, Regex) do
+    Enum.count(Regex.scan(pattern, string))
+  end
+
+  def count(string, pattern) do
+    Enum.count(:binary.matches(string, pattern))
```

Comment:
```suggestion
    length(Regex.scan(pattern, string, return: :index))
  end

  def count(string, pattern) do
    length(:binary.matches(string, pattern))
```

Reply:
I adjusted it to use `Kernel.length` instead, otherwise it would use the local string length function